service/service_test.go

```diff
@@ -137,3 +139,22 @@ var validServiceConfig = map[string]interface{}{
 func withConfig(data map[string]interface{}) Option {
 	return WithConfiguration(config.NewStaticProvider(data))
 }
+
+func TestAfterStartObserver(t *testing.T) {
+	t.Parallel()
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	h := &host{
+		serviceCore: serviceCore{
+			loggingCore: loggingCore{
+				log: ulog.NopLogger,
+			},
+		},
+		observer: AfterStart(func() {
+			wg.Done()
+		}),
+	}
+
+	h.StartAsync()
```

Comment:
This is just silly :)
I don't think we should test empty functions or functions always returning nils/true..